workers/ohsome_quality_analyst/indicators/currentness/indicator.py

```diff
@@ -0,0 +1,183 @@
+    ) -> None:
+        super().__init__(
+            layer_name=layer_name,
+            feature=feature,
+        )
+        self.threshold_yellow = 0.6
+        self.threshold_red = 0.2
+        self.element_count = None
+        self.contribution_sum = 0
+        self.contributions = {}
+        self.ratio = {}
+
+    async def preprocess(self) -> None:
+        latest_ohsome_stamp = await ohsome_client.get_latest_ohsome_timestamp()
+        start = "2008-01-01"
+        self.end = latest_ohsome_stamp.strftime("%Y-%m-%d")
+        time_range = "{0}/{1}/{2}".format(start, self.end, "P1Y")
+        curr_year_start = "{0}-01-01".format(latest_ohsome_stamp.year)
+        curr_year_range = "{0}/{1}".format(curr_year_start, self.end)
+
+        response = await ohsome_client.query(
+            layer=self.layer,
+            bpolys=self.feature.geometry,
+        )
+        self.element_count = response["result"][0]["value"]
+        self.result.timestamp_osm = dateutil.parser.isoparse(
+            response["result"][0]["timestamp"]
+        )
+        response_contributions = await ohsome_client.query(
+            layer=self.layer,
+            bpolys=self.feature.geometry,
+            time=time_range,
+            endpoint="contributions/latest/count",
+        )
+        for year in response_contributions["result"]:
+            time = dateutil.parser.isoparse(year["fromTimestamp"])
+            count = year["value"]
+            self.contributions[time.strftime("%Y")] = count
+
+        curr_year_response_contributions = await ohsome_client.query(
+            layer=self.layer,
+            bpolys=self.feature.geometry,
+            time=curr_year_range,
+            endpoint="contributions/latest/count",
+        )
+        time = dateutil.parser.isoparse(
+            curr_year_response_contributions["result"][0]["fromTimestamp"]
+        )
+        count = curr_year_response_contributions["result"][0]["value"]
+        self.contributions[time.strftime("%Y")] = count
+
+    def calculate(self) -> None:
+        logging.info(f"Calculation for indicator: {self.metadata.name}")
+
+        self.contribution_sum = sum(self.contributions.values())
+        # It can be that features are counted, but have been deleted since.
+        if self.element_count == 0:
+            self.result.description = (
+                "In the area of intrest no features "
+                "matching the filter are present today."
+            )
+            return
+        contributions_rel = self.contribution_sum
+        last_edited_year = ""
+        for year in self.contributions:
+            self.ratio[year] = (contributions_rel / self.contribution_sum) * 100
+            contributions_rel -= self.contributions[year]
+            self.contributions[year] = (
+                self.contributions[year] / self.contribution_sum
+            ) * 100
+            if self.contributions[year] != 0:
+                last_edited_year = year
```

Comment:
To add to the documentation topic. It would be nice to have somewhere a description how the result value of this indicator is calculated since it is a combination of two different parameter.

Reply:
I just remebered that there is already a description of the indicator in metadata.yaml. Maybe it is a good idea to copy this information to the docstring of the class?